libraries/AP_HAL_ChibiOS/hwdef/common/malloc.c

```diff
@@ -71,6 +80,19 @@ static memory_heap_t dma_reserve_heap;
  */
 void malloc_init(void)
 {
+#ifdef HAL_CHIBIOS_ENABLE_MALLOC_GUARD
+    chMtxObjectInit(&mem_mutex);
+#endif
+
+#if defined(STM32H7)
+    // zero first 1k of ITCM. We leave 1k free to avoid addresses
+    // close to nullptr being valid. Zeroing it here means we can
+    // check for changes which indicate a write to an uninitialised
+    // object.  We start at address 0x4 as writing the first word
+    // causes a fault
+    memset((void*)0x00000004, 0, 1020);
```

Comment:
So this is only able to track down issues like writing to a ptr near 0x0 but just not at 0x0 itself.

Reply:
Also is it valid to set value at 0x0 in H7 board?

Reply:
i get a hard fault if I try to write at 0x0. I vaguely remember there is a control register that enables/disables access to address 0, but I can't find it at the moment.

Reply:
it turns out we can read/write all but the first byte. I've changed the code to check address 0x1 to 0x3 as well